src/sparql/QuantityValuePatternBuilder.ts

```diff
@@ -0,0 +1,259 @@
+import PropertyValueRelation from '@/data-model/PropertyValueRelation';
+import ReferenceRelation from '@/data-model/ReferenceRelation';
+import UnitValue from '@/data-model/UnitValue';
+import { Condition } from '@/sparql/QueryRepresentation';
+import rdfNamespaces from '@/sparql/rdfNamespaces';
+import SyntaxBuilder from '@/sparql/SyntaxBuilder';
+import TripleBuilder from '@/sparql/TripleBuilder';
+import ValuePatternBuilder from '@/sparql/ValuePatternBuilder';
+import { BindPattern, Pattern } from 'sparqljs';
+
+export default class QuantityValuePatternBuilder implements ValuePatternBuilder {
+
+	private readonly tripleBuilder: TripleBuilder;
+
+	public constructor() {
+		this.tripleBuilder = new TripleBuilder();
+	}
+
+	public buildValuePatternFromCondition( condition: Condition, conditionIndex: number ): Pattern[] {
+		const {
+			propertyId,
+			negate,
+			referenceRelation,
+			propertyValueRelation,
+			datatype,
+			value,
+		} = condition;
+
+		if ( datatype !== 'quantity' ) {
+			throw new Error( 'Unexpected datatype: ' + datatype );
+		}
+		if ( !this.isQuantityValue( value ) ) {
+			throw new Error( 'Unexpected value: ' + JSON.stringify( value ) );
+		}
+
+		const numericValue = value.value;
+		const unit = value.unit;
+
+		let patterns: Pattern[];
+		if ( unit === null || propertyValueRelation === PropertyValueRelation.Regardless ) {
+			patterns = this.buildPatternsForUnitlessValue(
+				propertyId,
+				numericValue,
+				conditionIndex,
+				referenceRelation,
+				propertyValueRelation,
+			);
+		} else {
+			patterns = this.buildFullQuantityPattern(
+				numericValue,
+				unit,
+				conditionIndex,
+				propertyId,
+				referenceRelation,
+				propertyValueRelation,
+			);
+		}
+
+		if ( negate ) {
+			return [ {
+				type: 'minus',
+				patterns: patterns,
+			} ];
+		}
+		return patterns;
+	}
+
+	private buildFullQuantityPattern(
+		numericValue: number,
+		unit: string,
+		conditionIndex: number,
+		propertyId: string,
+		referenceRelation: ReferenceRelation,
+		propertyValueRelation: PropertyValueRelation,
+	): Pattern[] {
+		const patterns: Pattern[] = [];
+
+		const userQuantityVariable = SyntaxBuilder.buildVariableTermFromName( 'userQuantity' );
```

Comment:
Suggestion for refactoring for later. Inject SyntaxBuilder similar to TripleBuilder to make dependencies of this class clear.

Reply:
Done as a consequence of doing it for one of the previous commits in the chain